### PR TITLE
Preventuserspidjunctionput

### PR DESCRIPTION
**Pull request by noodlemoodle**
- removed SetUserServiceProviderJunctionRuleAction
- prevent user from adding ucjs with ServiceProvider/spid-specific capabilities where capability doesn't match their spid by
   - making Capability 'spid-aware'
   - create ucjdao/bareucjdao rule to prevent puts where there is a user capability spid mismatch
configure spids (not done) : https://github.com/nanoPayinc/NANOPAY/pull/13350

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
Looking at https://github.com/nanoPayinc/NANOPAY/pull/13350/files
Previously spid capabilities were dependents of a spid capability. Are we duplicating things now with both explicitly setting a spid on a capability and also capabilities under a spid capability?

**Comment by noodlemoodle** [comment]
> Looking at https://github.com/nanoPayinc/NANOPAY/pull/13350/files
> Previously spid capabilities were dependents of a spid capability. Are we duplicating things now with both explicitly setting a spid on a capability and also capabilities under a spid capability?

kind of, but there's nothing that explicitly says for example trevisoMenuCapability can't be granted to a user under 'bepay' spid, the only connection with it and the 'treviso' spid capability is the prerequisitecapabilityjunction itself
but i think we should have some way to distinguish when a capability such as this should only be granted to users under a certain spid, and the prerequisitecapabilityjunction wouldn't be sufficient since a spid capability might have other prerequisites that could be shared across multiple spids

**Comment by zahnno** [comment]
Instead of dealing with juggling ucjs, users, capabilities and writing multiple entries to journals - we could add an auth layer on the auth service that grabs the spid capability, its prerequisites and checks if the permission being checked for is included in the conglomerated permission list - this logic would be hit by both the check method and checkUser/Group method

File: src/foam/nanos/crunch/rules.jrl [hunk 1]
```diff
@@ -12,20 +12,32 @@ p({
...
-  "action":{
-    "class":"foam.nanos.auth.SetUserServiceProviderJunctionRuleAction"
-  },
+  "action":{ "class":"foam.nanos.auth.CheckUserCapabilitySpidMismatch" },
+  "lifecycleState":1
+})
+
+p({
+  "class":"foam.nanos.ruler.Rule",
+  "id":"68afcf0c-c718-98f8-0841-23sd-213d-1",
```
Comment (lchanmann) [inline src/foam/nanos/crunch/rules.jrl:32]: id is the same as the previous one?

**Pull request closed without merging**